lib/ansible/modules/cloud/azure/azure_rm_webapp.py

```diff
@@ -536,17 +537,29 @@ def exec_module(self, **kwargs):
                         self.fail('Unsupported framework {0} for Linux web app.'.format(self.frameworks[0]['name']))
 
                     self.site_config['linux_fx_version'] = (self.frameworks[0]['name'] + '|' + self.frameworks[0]['version']).upper()
+
+                    if self.frameworks[0]['name'] == 'java':
+                        if self.frameworks[0]['version'] != '8':
+                            self.fail("Linux web app only supports java 8.")
+                        if self.frameworks[0]['settings'] and self.frameworks[0]['settings']['java_container'].lower() != 'tomcat':
+                            self.fail("Linux web app only supports tomcat container.")
+
+                        if self.frameworks[0]['settings'] and self.frameworks[0]['settings']['java_container'].lower() == 'tomcat':
+                            self.site_config['linux_fx_version'] = 'TOMCAT|' + self.frameworks[0]['settings']['java_container_version'] + '-jre8'
+                        else:
+                            self.site_config['linux_fx_version'] = 'JAVA|8-jre8'
                 else:
                     for fx in self.frameworks:
                         if fx.get('name') not in self.supported_windows_frameworks:
                             self.fail('Unsupported framework {0} for Windows web app.'.format(fx.get('name')))
                         else:
                             self.site_config[fx.get('name') + '_version'] = fx.get('version')
 
-                for fx in self.frameworks:
-                    if 'settings' in fx and fx['settings'] is not None:
-                        for key, value in fx['settings'].items():
-                            self.site_config[key] = value
+                if not is_linux:
```

Comment:
could this could just go under else: above?

Reply:
it's not right. currently, only java_container and java_contatiner_setting in settings, which is not applied to linux web app. but it's  not true for other settings added in future.  so seperate the logic  to check settings.

Reply:
I mean there's:

if is_linux:
...
else:

....

if not is_linux:

...
or I mismatched levels?